config/config.go

```diff
@@ -13,13 +17,33 @@ type Config struct {
 	Transport http.RoundTripper // Custom HTTP transport, allowing optional customization of the HTTP client behavior.
 }
 
-// NewDefaultConfig returns a default configuration for connecting to the SPV Wallet API,
-// setting a one-minute timeout, using the default HTTP transport, and applying the
-// base API address as the addr value.
-func NewDefaultConfig(addr string) Config {
-	return Config{
-		Addr:      addr,
-		Timeout:   1 * time.Minute,
-		Transport: http.DefaultTransport,
+// New creates a new Config instance with optional customizations.
+func New(options ...Option) Config {
+	cfg := Config{}
+	for _, opt := range options {
+		opt(&cfg)
+	}
+	if err := cfg.Validate(); err != nil {
```

Comment:
Maybe I'm wrong, but this should be rather after setDefaultValues, 
otherwise, if you won't provide all the options with some value, the program will fail and setDefaultValues actually won't be ever needed

Reply:
do we need to check default values too ?

Reply:
good point